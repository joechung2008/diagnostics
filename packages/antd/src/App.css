html,
body {
  height: 100%;
  margin: 0;
  padding: 0;
}

*,
*::after,
*::before {
  box-sizing: border-box;
}

.centered {
  align-items: center;
  display: flex;
  height: 100%;
  justify-content: center;
  width: 100%;
}

.extension-root {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
}

.flexbox {
  display: flex;
  flex-direction: column;
  height: 100vh;
}

.grow {
  flex: 1;
}

.stack {
  display: flex;
  flex-direction: row;
  height: 100%;
}

.tab-list {
  padding-left: 1rem;
}

.tab-panel {
  flex: 1;
  overflow-y: auto;
}
