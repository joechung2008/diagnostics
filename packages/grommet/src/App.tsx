import {
  Button,
  Menu,
  Notification,
  Spinner,
  Tab,
  Tabs,
  Toolbar,
} from "grommet";
import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo } from "./utils";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

const App: React.FC = () => {
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<string>(Environment.Public);
  const [activeIndex, setActiveIndex] = useState<number>(0);

  const environmentName = useMemo(() => {
    switch (environment) {
      case Environment.Public:
        return "Public Cloud";
      case Environment.Fairfax:
        return "Fairfax";
      case Environment.Mooncake:
        return "Mooncake";
      default:
        return "Select environment";
    }
  }, [environment]);

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR(environment, async (url) => {
    const response = await fetch(url);
    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === Environment.Public,
        onClick: () => {
          setEnvironment(Environment.Public);
          setExtension(undefined);
          setActiveIndex(0);
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === Environment.Fairfax,
        onClick: () => {
          setEnvironment(Environment.Fairfax);
          setExtension(undefined);
          setActiveIndex(0);
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === Environment.Mooncake,
        onClick: () => {
          setEnvironment(Environment.Mooncake);
          setExtension(undefined);
          setActiveIndex(0);
        },
      },
    ],
    [environment]
  );

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className="flexbox">
      <Toolbar align="center">
        <Menu
          label={environmentName}
          items={environments.map((env) => ({
            label: env.text,
            onClick: env.onClick,
          }))}
        />
        <Activity mode={showPaasServerless ? "visible" : "hidden"}>
          <Button
            label="paasserverless"
            plain
            onClick={() => {
              const paasserverless = diagnostics?.extensions["paasserverless"];
              if (isExtensionInfo(paasserverless)) {
                setExtension(paasserverless);
                setActiveIndex(0);
              }
            }}
          />
        </Activity>
        <Button
          label="websites"
          plain
          onClick={() => {
            const websites = diagnostics?.extensions["websites"];
            if (isExtensionInfo(websites)) {
              setExtension(websites);
              setActiveIndex(0);
            }
          }}
        />
      </Toolbar>
      <Tabs activeIndex={activeIndex} justify="start" onActive={setActiveIndex}>
        <Tab title="Extensions" />
        <Tab title="Build Information" />
        <Tab title="Server Information" />
      </Tabs>
      <Activity mode={!error && isLoading ? "visible" : "hidden"}>
        <div className="tab-panel">
          <div className="centered">
            <Spinner aria-label="Loading..." />
          </div>
        </div>
      </Activity>
      <Activity mode={error && !isLoading ? "visible" : "hidden"}>
        <div className="tab-panel">
          <div className="centered">
            <Notification
              status="critical"
              message={`Error: ${error instanceof Error ? error.message : "Unknown error"}`}
            />
          </div>
        </div>
      </Activity>
      <Activity
        mode={!error && !isLoading && activeIndex === 0 ? "visible" : "hidden"}
      >
        {diagnostics?.extensions && (
          <div className="tab-panel">
            <div className="stack">
              <Extensions
                extensions={diagnostics.extensions}
                onLinkClick={handleLinkClick}
              />
              {extension && <Extension {...extension} />}
            </div>
          </div>
        )}
      </Activity>
      <Activity
        mode={!error && !isLoading && activeIndex === 1 ? "visible" : "hidden"}
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <BuildInfo {...diagnostics.buildInfo} />
          </div>
        )}
      </Activity>
      <Activity
        mode={!error && !isLoading && activeIndex === 2 ? "visible" : "hidden"}
      >
        {diagnostics?.serverInfo && (
          <div className="tab-panel">
            <ServerInfo {...diagnostics.serverInfo} />
          </div>
        )}
      </Activity>
    </div>
  );
};

export default App;
