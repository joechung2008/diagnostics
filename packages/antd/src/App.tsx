import { Empty, Flex, Menu, Spin, Tabs } from "antd";
import { Activity, useCallback, useMemo, useState } from "react";
import useSWR from "swr";
import BuildInfo from "./BuildInfo";
import Extension from "./Extension";
import Extensions from "./Extensions";
import ServerInfo from "./ServerInfo";
import { isExtensionInfo, when } from "./utils";

import "./App.css";

const Environment = {
  Public: "https://hosting.portal.azure.net/api/diagnostics",
  Fairfax: "https://hosting.azureportal.usgovcloudapi.net/api/diagnostics",
  Mooncake: "https://hosting.azureportal.chinacloudapi.cn/api/diagnostics",
} as const;

type EnvironmentUrl = (typeof Environment)[keyof typeof Environment];

const items = [
  { key: "extensions", label: "Extensions" },
  { key: "build", label: "Build Information" },
  { key: "server", label: "Server Information" },
];

const App: React.FC = () => {
  const [extension, setExtension] = useState<ExtensionInfo>();
  const [environment, setEnvironment] = useState<EnvironmentUrl>(
    Environment.Public
  );
  const [selectedTab, setSelectedTab] = useState("extensions");

  const {
    data: diagnostics,
    error,
    isLoading,
  } = useSWR<Diagnostics>(environment, async (environment: string) => {
    const response = await fetch(environment);

    if (!response.ok) {
      throw new Error(`Error fetching diagnostics: ${response.statusText}`);
    }

    return response.json();
  });

  const environmentName = useMemo(() => {
    switch (environment) {
      case Environment.Public:
        return "Public Cloud";
      case Environment.Fairfax:
        return "Fairfax";
      case Environment.Mooncake:
        return "Mooncake";
      default:
        return "Select environment";
    }
  }, [environment]);

  const showPaasServerless = useMemo(
    () => isExtensionInfo(diagnostics?.extensions["paasserverless"]),
    [diagnostics?.extensions]
  );

  const environments = useMemo(
    () => [
      {
        key: "public",
        text: "Public Cloud",
        selected: environment === Environment.Public,
        onClick: () => {
          setEnvironment(Environment.Public);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "fairfax",
        text: "Fairfax",
        selected: environment === Environment.Fairfax,
        onClick: () => {
          setEnvironment(Environment.Fairfax);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
      {
        key: "mooncake",
        text: "Mooncake",
        selected: environment === Environment.Mooncake,
        onClick: () => {
          setEnvironment(Environment.Mooncake);
          setExtension(undefined);
          setSelectedTab("extensions");
        },
      },
    ],
    [environment]
  );

  const menuItems = useMemo(
    () => [
      {
        key: "env-dropdown",
        label: environmentName,
        children: environments.map((env) => ({
          key: env.key,
          label: env.text,
          onClick: env.onClick,
        })),
      },
      ...when(showPaasServerless, {
        key: "paasserverless",
        label: "paasserverless",
        onClick: () => {
          const paasserverless = diagnostics?.extensions["paasserverless"];
          if (isExtensionInfo(paasserverless)) {
            setExtension(paasserverless);
            setSelectedTab("extensions");
          }
        },
      }),
      {
        key: "websites",
        label: "websites",
        onClick: () => {
          const websites = diagnostics?.extensions["websites"];
          if (isExtensionInfo(websites)) {
            setExtension(websites);
            setSelectedTab("extensions");
          }
        },
      },
    ],
    [environments, environmentName, showPaasServerless, diagnostics?.extensions]
  );

  const handleLinkClick = useCallback(
    (_?: React.MouseEvent, item?: KeyedNavLink) => {
      if (item) {
        const extension = diagnostics?.extensions[item.key];
        if (isExtensionInfo(extension)) {
          setExtension(extension);
        }
      }
    },
    [diagnostics?.extensions]
  );

  return (
    <div className="flexbox">
      <Menu mode="horizontal" items={menuItems} />
      <Tabs
        activeKey={selectedTab}
        className="tab-list"
        items={items}
        onChange={(key) => setSelectedTab(key)}
      />
      <Activity mode={isLoading ? "visible" : "hidden"}>
        <div className="tab-panel">
          <Flex className="centered">
            <Spin size="large" />
          </Flex>
        </div>
      </Activity>
      <Activity mode={error && !isLoading ? "visible" : "hidden"}>
        {error && (
          <div className="tab-panel">
            <Flex className="centered">
              <Empty
                description={`Error: ${error.message ?? "Unexpected error"}`}
              />
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "extensions"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.extensions && (
          <div className="tab-panel">
            <Flex className="stack">
              <Extensions
                extensions={diagnostics.extensions}
                onLinkClick={handleLinkClick}
              />
              {extension && <Extension {...extension} />}
            </Flex>
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "build" ? "visible" : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <BuildInfo {...diagnostics.buildInfo} />
          </div>
        )}
      </Activity>
      <Activity
        mode={
          !error && !isLoading && selectedTab === "server"
            ? "visible"
            : "hidden"
        }
      >
        {diagnostics?.buildInfo && (
          <div className="tab-panel">
            <ServerInfo {...diagnostics.serverInfo} />
          </div>
        )}
      </Activity>
    </div>
  );
};

export default App;
