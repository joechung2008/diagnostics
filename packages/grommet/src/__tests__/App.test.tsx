import { render } from "@testing-library/react";
import { beforeEach, describe, expect, it, vi, type Mock } from "vitest";
import * as swr from "swr";
import App from "../App";

const mockDiagnostics = {
  buildInfo: {},
  extensions: {
    websites: { extensionName: "Websites" },
    paasserverless: { extensionName: "PaaS Serverless" },
  },
  serverInfo: {
    deploymentId: "test-deployment",
    extensionSync: { totalSyncAllCount: 42 },
    hostname: "test-host",
    nodeVersions: "v18",
    serverId: "test-server",
    uptime: 86400,
  },
};

vi.mock("swr", () => ({
  default: vi.fn(),
}));

beforeEach(() => {
  (swr.default as Mock).mockReturnValue({
    data: mockDiagnostics,
    error: undefined,
    isLoading: false,
  });
  global.scrollTo = vi.fn();
});

describe("App", () => {
  it("renders nothing while loading", () => {
    (swr.default as Mock).mockReturnValueOnce({
      data: undefined,
      error: undefined,
      isLoading: true,
    });
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders error message on fetch error", () => {
    (swr.default as Mock).mockReturnValueOnce({
      data: undefined,
      error: new Error("Network error"),
      isLoading: false,
    });
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it("renders Extensions tab by default", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
