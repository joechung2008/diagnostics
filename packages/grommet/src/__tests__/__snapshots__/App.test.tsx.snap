// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders Extensions tab by default 1`] = `
<div>
  <div
    class="flexbox"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 dLliyK"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Open Menu"
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 kYyYKN"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jTcNpm"
          >
            Public Cloud
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kOnJGZ"
          />
          <svg
            aria-label="FormDown"
            class="StyledIcon-sc-ofa7kd-0 krSgpg"
            viewBox="0 0 24 24"
          >
            <path
              d="m18 9-6 6-6-6"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 fDLUqh StyledTabs-sc-a4fwxl-2 lfIHdT"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fBuFhc"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dOjkO StyledTabs__StyledTabsHeader-sc-a4fwxl-0 gBVpTp"
          role="tablist"
        >
          <button
            aria-expanded="true"
            aria-selected="true"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 edndUX StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 elmtiN"
              >
                Extensions
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Build Information
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Server Information
              </span>
            </div>
          </button>
        </div>
      </div>
      <div
        aria-label="Extensions Tab Contents"
        class="StyledTabs__StyledTabPanel-sc-a4fwxl-1 kwlwid"
        role="tabpanel"
      />
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <div
        class="centered"
      >
        <div
          aria-label="Loading..."
          class="StyledBox-sc-13pk1d4-0 ikriZb"
        />
      </div>
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <div
        class="centered"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 ehpCpY"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 hidwNH"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 gOfGFJ"
            >
              <svg
                aria-label="Status is critical"
                class="StyledIcon-sc-ofa7kd-0 eBAFuY"
                viewBox="0 0 12 12"
              >
                <path
                  d="M6.712 1.263a1.005 1.005 0 0 0-1.424 0L1.263 5.288a1.005 1.005 0 0 0 0 1.424l4.025 4.025a1.005 1.005 0 0 0 1.424 0l4.025-4.025a1.005 1.005 0 0 0 0-1.424L6.712 1.263z"
                  fill-rule="evenodd"
                  stroke="#000"
                />
              </svg>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 fDLUqh"
            >
              <p
                class="StyledParagraph-sc-tbetod-0 dNUDyP"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 gKmPWp"
                >
                  Error: Unknown error
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tab-panel"
    >
      <div
        class="stack"
      >
        <nav
          aria-label="Extensions"
          class="StyledBox-sc-13pk1d4-0 eznImF extension-root overflow-x-hidden"
        >
          <button
            class="StyledButton-sc-323bzc-0 jLPXpV"
            type="button"
          >
            PaaS Serverless
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hrJqFx"
          />
          <button
            class="StyledButton-sc-323bzc-0 jLPXpV"
            type="button"
          >
            Websites
          </button>
        </nav>
      </div>
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <table
        aria-label="Build Info"
        class="StyledTable-sc-1m3u5g-6 jlczSm"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kSxzat"
              scope="col"
            >
              Name
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kSxzat"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Build Version
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            />
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <table
        aria-label="Server Info"
        class="StyledTable-sc-1m3u5g-6 jlczSm"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kSxzat"
              scope="col"
            >
              Name
            </th>
            <th
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kSxzat"
              scope="col"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Hostname
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              test-host
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Uptime
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              86400
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Server ID
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              test-server
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Deployment ID
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              test-deployment
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Node Versions
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              v18
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              Extension Sync | Total Sync All Count
            </td>
            <td
              class="StyledTable__StyledTableCell-sc-1m3u5g-0 kCLlzz"
            >
              42
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`App > renders error message on fetch error 1`] = `
<div>
  <div
    class="flexbox"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 dLliyK"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Open Menu"
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 kYyYKN"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jTcNpm"
          >
            Public Cloud
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kOnJGZ"
          />
          <svg
            aria-label="FormDown"
            class="StyledIcon-sc-ofa7kd-0 krSgpg"
            viewBox="0 0 24 24"
          >
            <path
              d="m18 9-6 6-6-6"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        style="display: none;"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 fDLUqh StyledTabs-sc-a4fwxl-2 lfIHdT"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fBuFhc"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dOjkO StyledTabs__StyledTabsHeader-sc-a4fwxl-0 gBVpTp"
          role="tablist"
        >
          <button
            aria-expanded="true"
            aria-selected="true"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 edndUX StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 elmtiN"
              >
                Extensions
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Build Information
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Server Information
              </span>
            </div>
          </button>
        </div>
      </div>
      <div
        aria-label="Extensions Tab Contents"
        class="StyledTabs__StyledTabPanel-sc-a4fwxl-1 kwlwid"
        role="tabpanel"
      />
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <div
        class="centered"
      >
        <div
          aria-label="Loading..."
          class="StyledBox-sc-13pk1d4-0 ikriZb"
        />
      </div>
    </div>
    <div
      class="tab-panel"
    >
      <div
        class="centered"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 ehpCpY"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 hidwNH"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 gOfGFJ"
            >
              <svg
                aria-label="Status is critical"
                class="StyledIcon-sc-ofa7kd-0 eBAFuY"
                viewBox="0 0 12 12"
              >
                <path
                  d="M6.712 1.263a1.005 1.005 0 0 0-1.424 0L1.263 5.288a1.005 1.005 0 0 0 0 1.424l4.025 4.025a1.005 1.005 0 0 0 1.424 0l4.025-4.025a1.005 1.005 0 0 0 0-1.424L6.712 1.263z"
                  fill-rule="evenodd"
                  stroke="#000"
                />
              </svg>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 fDLUqh"
            >
              <p
                class="StyledParagraph-sc-tbetod-0 dNUDyP"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 gKmPWp"
                >
                  Error: Network error
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App > renders nothing while loading 1`] = `
<div>
  <div
    class="flexbox"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 dLliyK"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="Open Menu"
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 kYyYKN"
        >
          <span
            class="StyledText-sc-1sadyjn-0 jTcNpm"
          >
            Public Cloud
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kOnJGZ"
          />
          <svg
            aria-label="FormDown"
            class="StyledIcon-sc-ofa7kd-0 krSgpg"
            viewBox="0 0 24 24"
          >
            <path
              d="m18 9-6 6-6-6"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </div>
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        style="display: none;"
        type="button"
      >
        paasserverless
      </button>
      <button
        class="StyledButton-sc-323bzc-0 jLPXpV"
        type="button"
      >
        websites
      </button>
    </div>
    <div
      class="StyledBox-sc-13pk1d4-0 fDLUqh StyledTabs-sc-a4fwxl-2 lfIHdT"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 fBuFhc"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 dOjkO StyledTabs__StyledTabsHeader-sc-a4fwxl-0 gBVpTp"
          role="tablist"
        >
          <button
            aria-expanded="true"
            aria-selected="true"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 edndUX StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 elmtiN"
              >
                Extensions
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Build Information
              </span>
            </div>
          </button>
          <button
            aria-expanded="false"
            aria-selected="false"
            class="StyledButton-sc-323bzc-0 jLPXpV"
            role="tab"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bWNHnf StyledTab-sc-1nnwnsb-0 dIJknn"
            >
              <span
                class="StyledText-sc-1sadyjn-0 bUOfNz"
              >
                Server Information
              </span>
            </div>
          </button>
        </div>
      </div>
      <div
        aria-label="Extensions Tab Contents"
        class="StyledTabs__StyledTabPanel-sc-a4fwxl-1 kwlwid"
        role="tabpanel"
      />
    </div>
    <div
      class="tab-panel"
    >
      <div
        class="centered"
      >
        <div
          aria-label="Loading..."
          class="StyledBox-sc-13pk1d4-0 ikriZb"
        />
      </div>
    </div>
    <div
      class="tab-panel"
      style="display: none;"
    >
      <div
        class="centered"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 ehpCpY"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 hidwNH"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 gOfGFJ"
            >
              <svg
                aria-label="Status is critical"
                class="StyledIcon-sc-ofa7kd-0 eBAFuY"
                viewBox="0 0 12 12"
              >
                <path
                  d="M6.712 1.263a1.005 1.005 0 0 0-1.424 0L1.263 5.288a1.005 1.005 0 0 0 0 1.424l4.025 4.025a1.005 1.005 0 0 0 1.424 0l4.025-4.025a1.005 1.005 0 0 0 0-1.424L6.712 1.263z"
                  fill-rule="evenodd"
                  stroke="#000"
                />
              </svg>
            </div>
            <div
              class="StyledBox-sc-13pk1d4-0 fDLUqh"
            >
              <p
                class="StyledParagraph-sc-tbetod-0 dNUDyP"
              >
                <span
                  class="StyledText-sc-1sadyjn-0 gKmPWp"
                >
                  Error: Unknown error
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
